src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +578,19 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        if (seenReadonly)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (previousModifier?.Kind() != SyntaxKind.RefKeyword)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_RefReadOnlyWrongOrdering, modifier);
+                        }
+                        else if (seenRef)
+                        {
+                            Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefReadonlyParameters, diagnostics);
+                            seenReadonly = true;
```

Comment:
Should we unconditionally set `seenReadonly`?

Reply:
What is the reason to set `seenReadonly` conditionally?

Reply:
> What is the reason to set seenReadonly conditionally?

It does not look useful to set it in a any other condition. Other cases do not look at the value. An error that `readonly` it is not preceded by `ref` is much better than an error about a duplicate, in my opinion.

Reply:
FWIW, this is the difference: https://github.com/dotnet/roslyn/commit/8cdd5410f44aa11cccb2e87225be24f1d89a6e36

Reply:
Basically `readonly` means nothing unless it is part of `ref readonly` and the ref is actually not ignored.